### PR TITLE
test: add MouseEventManager edge cases and error handling tests

### PR DESCRIPTION
**Pull request by MasatoMakino**
## Summary

This PR adds a new test suite for MouseEventManager edge cases and error handling, along with enhanced test infrastructure for boundary condition testing.

**New Test File**: `__test__/MouseEventManager.edge-cases.spec.ts`
- Tests MouseEventManager's robustness under error conditions and edge cases
- Focuses on conditions that could cause system instability or unexpected behavior in production environments

**Enhanced Test Infrastructure**: `MouseEventManagerScene`
- Added configurable canvas dimensions for boundary condition testing  
- Supports custom canvas width/height while maintaining backward compatibility

## Test Categories

- **Error-prone Object Handling**: currentOver state management during object removal
- **Invalid Event Processing**: malformed events, invalid coordinates, stress testing
- **System Boundary Conditions**: zero-size canvas, empty scenes
- **Robustness Verification**: complex error scenarios, state consistency

## Test Scope & Rationale

This project operates within TypeScript's type safety guarantees and browser JavaScript execution environment. The tests focus on application-layer error handling and recovery, configuration validity, and integration with Three.js and DOM APIs under adverse conditions. Testing scenarios that assume fundamental system failures (JavaScript engine corruption, browser bugs, OS failures) would lead to infinite test expansion and maintenance complexity without practical value for the project's actual use cases.

**Explicitly Not Tested** (Out of Project Scope):
- TypeScript type system validation (e.g., `typeof` checks for primitives)
- JavaScript engine memory corruption or low-level runtime failures
- Browser implementation bugs or inconsistencies
- Operating system or hardware-level failures
- Prototype pollution attacks or malicious code injection

## Test Plan

- [x] All existing tests continue to pass (309 tests passed)
- [x] New edge case tests validate error handling robustness (7 new tests)
- [x] Enhanced test infrastructure maintains backward compatibility
- [x] Code quality checks pass

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

- Tests
  - Added comprehensive edge-case test suite for mouse interaction management, covering invalid inputs, object removal during interaction, empty scenes, and zero-size canvases to ensure stability and proper cleanup.
  - Introduced configurable test scene dimensions to simulate varied display scenarios and verify aspect-ratio handling.
  - Strengthened robustness checks for state coherence under rapid, repeated events and dynamic scene changes, reducing risk of crashes and regressions.

### CODE REVIEW

File: __test__/MouseEventManagerScene.ts [hunk 1]
```diff
@@ -50,38 +66,47 @@ export class MouseEventManagerScene {
...
+    } = options || {};
+
     this.scene = new Scene();
-    this.camera = new PerspectiveCamera(
-      45,
-      MouseEventManagerScene.W / MouseEventManagerScene.H,
-      1,
-      400,
-    );
+    this.camera = new PerspectiveCamera(45, canvasWidth / canvasHeight, 1, 400);
```
[the remark this answers is not in the document]
Comment (MasatoMakino) [reply __test__/MouseEventManagerScene.ts:93]: This division by zero is intentional for edge case testing. See test "should handle zero-size canvas initialization and coordinate processing gracefully" in  __test__/MouseEventManager.edge-cases.spec.ts:263 which explicitly creates zero-size  canvas (canvasWidth: 0, canvasHeight: 0) to validate MouseEventManager's robustness  under corrupted camera projection matrices (aspect ratio = Infinity/NaN). 
Fixing this  would eliminate the test's value in proving the system handles extreme boundary conditions gracefully.
